joomla/cassiopeia
# adds ability to have scrollable back to top

Pull Request for Issue # .

### Summary of Changes
Adds ability for "smart" back to top where...
1) back to top = on & scroll = off  : button is in footer within container.
2) back to top = off & scroll ( doesn't matter )  : no button displayed
3) back to top = on & scroll = on : button moved out of container and scrolls with viewport.
button is hidden until content is scrolled.


### Testing Instructions
check that backtotop button works in footer and turns on/off
load pr
run npm ci
check that button states are as listed above.


### Expected result
**backend**
![image](https://user-images.githubusercontent.com/1850089/94995791-3772b700-0566-11eb-8c04-7c91b65eb0fc.png)

**desktop**
![image](https://user-images.githubusercontent.com/1850089/94995551-b36bff80-0564-11eb-8ed5-d2958a084a6f.png)

**iphone11**
![image](https://user-images.githubusercontent.com/1850089/94995658-49078f00-0565-11eb-86ce-ded293c0b7c8.jpg)

**note8**
![image](https://user-images.githubusercontent.com/1850089/94995659-49a02580-0565-11eb-8b15-000bed189603.jpg

### Actual result
![image](https://user-images.githubusercontent.com/1850089/94995670-6c323e80-0565-11eb-8620-7e52f262f880.png)
![image](https://user-images.githubusercontent.com/1850089/94995674-76543d00-0565-11eb-95ba-f5dbfce429e3.png)

### Documentation Changes Required
none
